ichnaea/data/tests/test_station.py

```diff
@@ -137,10 +151,150 @@ def mock_area(self, updated_areas, key, num=num, session2=session2):
                 )
                 == 1
             )
+            assert metricsmock.has_record(
+                "incr",
+                "data.station.dberror",
+                value=1,
+                tags=["type:cell", "errno:%s" % LOCK_WAIT_TIMEOUT],
+            )
         finally:
             CellUpdater._retry_wait = orig_wait
             session.execute(text("drop table %s;" % cell.__tablename__))
 
+    @pytest.mark.parametrize(
+        "errclass,errno,errmsg",
+        (
+            (
+                # This was an InternalError before PyMySQL 0.9.0
+                InternalError,
+                LOCK_DEADLOCK,
+                "Deadlock found when trying to get lock; try restarting transaction",
+            ),
```

Comment:
We don't need to support PyMySQL versions we're not using. Why is this test helpful to have going forward?

If we're concerned about LOCK_DEADLOCK being switched to something different, I feel like we should do a different kind of test here, but I can't think of a way to do that without it being really ornery which is its own kind of maintenance struggle.

Reply:
That's a good point, I'll drop the legacy test case.